### PR TITLE
Add support for meliae, objdump, manhole and a bunch of fixes and improvements

### PR DESCRIPTION
**Pull request by whirm**

### CODE REVIEW

**Comment by whirm** [comment]
@corpaul ready to merge

**Comment by whirm** [comment]
everyone is out, so I'm merging it myself

**Comment by egbertbouman** [comment]
@whirm It looks like the changes to the proccess guard broke the Jenkins PR tester again.

File: scripts/process_guard.py [hunk 1]
```diff
@@ -134,8 +135,10 @@ def run(self, cmd):
             stdout = stderr = None
 
         print >> stdout, "Starting #%05d: %s" % (self.cmd_counter, cmd)
+        stdout.flush()
```
Comment (egbertbouman) [inline scripts/process_guard.py:138]: This goes wrong when stdout == None.

**Comment by egbertbouman** [comment]
@whirm Nevermind, you already fixed it.

**Pull request merged**